src/gui/application.cpp

```diff
@@ -646,31 +549,106 @@ static void displayHelpText(const QString &t, std::ostream &stream = std::cout)
+    auto logDirOption = addOption({ "logdir", "Write each sync log output in a new file in folder.", "name" });
+    auto logExpireOption = addOption({ "logexpire", "Remove logs older than <hours> hours (to be used with --logdir).", "hours" });
+    auto logFlushOption = addOption({ "logflush", "Flush the log file after every write." });
+    auto logDebugOption = addOption({ "logdebug", "Output debug-level messages in the log." });
+    auto logWindowOption = addOption({ "logwindow", "//TODO//" });
+    auto languageOption = addOption({ "language", "Override UI language.", "language" });
+    auto confDirOption = addOption({ "confdir", "Use the given configuration folder.", "dirname" });
+    auto debugOption = addOption({ "debug", "Enable debug mode." });
+
+    // virtual file system parameters (optional)
+    parser.addPositionalArgument("vfs file", "Virtual file system file to be opened (optional).", { "[<vfs file>]" });
+
+    if (!parser.parse(arguments)) {
+        showHint(parser.errorText());
+    }
 
-void Application::showVersion()
-{
-    displayHelpText(Theme::instance()->versionSwitchOutput());
+    if (parser.isSet(helpOption)) {
+        showHint(parser.helpText());
+    }
+
+    if (parser.isSet(versionOption)) {
+        displayHelpText(_theme->versionSwitchOutput());
+        ::exit(0);
+    }
+
+    // TODO: rename this option (see #8234 for more information)
+    if (parser.isSet(showSettingsOption)) {
+        _showSettings = true;
+    }
+    if (parser.isSet(quitInstanceOption)) {
+        _quitInstance = true;
+    }
+    if (parser.isSet(logWindowOption)) {
+        _showLogWindow = true;
+    }
+    if (parser.isSet(logFileOption)) {
+        _logFile = parser.value(logFileOption);
+    }
+    if (parser.isSet(logDirOption)) {
+        _logDir = parser.value(logDirOption);
+    }
+    if (parser.isSet(logExpireOption)) {
+        _logExpire = std::chrono::hours(parser.value(logExpireOption).toInt());
+    }
+    if (parser.isSet(logFlushOption)) {
+        _logFlush = true;
+    }
+    if (parser.isSet(logDebugOption)) {
+        _logDebug = true;
+    }
+    if (parser.isSet(confDirOption)) {
+        const auto confDir = parser.value(confDirOption);
+        if (!ConfigFile::setConfDir(confDir)) {
+            showHint("Invalid path passed to --confdir");
+        }
+    }
+    if (parser.isSet(debugOption)) {
+        _logDebug = true;
+        _debugMode = true;
+    }
+    if (parser.isSet(languageOption)) {
+        _userEnforcedLanguage = parser.value(languageOption);
+    }
+
+    auto positionalArguments = parser.positionalArguments();
+
+    // ignore any positional arguments beyond the first one
+    if (!positionalArguments.empty()) {
+        QTimer::singleShot(0, this, [this, positionalArguments] { openVirtualFile(positionalArguments.front()); });
+    }
 }
 
 void Application::showHint(const QString &errorHint)
```

Comment:
Hmm maybe print the hint to the console and call showHelp()

Reply:
Well, yes or no? I don't mind having this method, but I don't mind removing it either.

Reply:
Removed the function in the latest commit, as discussed recently.